### PR TITLE
feat: Form dialog generation

### PR DESCRIPTION
**Pull request by hatpick**
## Description

Re-brands dialog-generation-flow as form-dialogs.
Fixes minor bugs.
Makes calls to server API for generation.

## Task Item

#minor
#3791 
#3792 

## Screenshots
form-dialogs1

### CODE REVIEW

File: Composer/packages/client/src/store/action/index.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 export * from './appUpdate';
 export * from './dialog';
+export * from './formDialog';
```
Comment (GeoffCoxMSFT) [inline Composer/packages/client/src/store/action/index.ts:6]: More index files.....

File: Composer/packages/extensions/form-dialogs/src/app/@types/index.d.ts [hunk 2]
```diff
@@ -1,16 +1,14 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (GeoffCoxMSFT) [inline Composer/packages/extensions/form-dialogs/src/app/@types/index.d.ts:4]: Consider doing this as a single line suppression to discourage future use of any in this file.

File: Composer/packages/extensions/form-dialogs/src/app/dispatcher/dispatcherUtil.ts [hunk 3]
```diff
@@ -11,7 +12,7 @@ export type HandlerCreator<TD = {}> = (dependencies: TD) => Record<string, Handl
 
 export const getDispatcher = <TD, TA extends Record<string, Handler>>(dependencies: (lifetime: ILifetime) => TD) => {
   const handlerMap = new Map<string, Handler>();
-  const handlerLifetimeMap = new WeakMap<HandlerCreator, Lifetime>();
+  const handlerLifetimeMap = new WeakMap<HandlerCreator<TD>, Lifetime>();
```
Comment (GeoffCoxMSFT) [inline Composer/packages/extensions/form-dialogs/src/app/dispatcher/dispatcherUtil.ts:15]: What do TD and TA stand for?

File: Composer/packages/extensions/form-dialogs/src/app/@types/index.d.ts (hunk 2 above)
Comment (hatpick) [reply Composer/packages/extensions/form-dialogs/src/app/@types/index.d.ts:4]: we don't even need it anymore

File: Composer/packages/extensions/form-dialogs/src/app/dispatcher/dispatcherUtil.ts (hunk 3 above)
Comment (hatpick) [reply Composer/packages/extensions/form-dialogs/src/app/dispatcher/dispatcherUtil.ts:15]: Dependency and Action

**Pull request merged**